### ISSUE TITLE
"Invalid frob" and Invalid auth token

### ISSUE THREAD

**Issue by cicciopizza** [trickortweak/flickr-uploader#86]
I created a new flickr account.
I asked and got a new API Key.
I have configured the uploadr.ini file by entering apy-key, secret and FILES_DIR: 

```
################################################################################
# Location to scan for new files
################################################################################
FILES_DIR = "/home/alfredo/Scaricati/flickr-uploader-master/foto"

################################################################################
#   Flickr settings
################################################################################
#   Set your own API key and secret message
#   Go to http://www.flickr.com/services/apps/create/apply and apply for an API key
#
FLICKR = {
        "title"                 : "",
        "description"           : "",
        "tags"                  : "auto-upload",
        "is_public"             : "0",
        "is_friend"             : "0",
        "is_family"             : "0",
        "api_key"               : "******19ae2e9c5add3e2fa4fb7bb6df",
        "secret"                : "******09f8d8f4bd"
        }
```
When I launch ./uploadr.py I get a message telling me to authenticate the application:

```
--------- Start time: Sat Jun 10 16:37:36 2017 ---------
False
Setting up the database: ./upload_flickr_db
Adding last_modified column to database
Completed database setup
Getting new token
Copy-paste following URL into a web browser and follow instructions:
https://api.flickr.com/services/auth/?perms=delete&api_key=******19ae2e9c5add3e2fa4fb7bb6df&frob=72157681960285453-b50078e0cf3fc638-930130&api_sig=******03f4d81639ec13fe4c8089206e
Have you authenticated this application? (Y/N): È stata creata una nuova finestra nella sessione corrente del browser.
```

A new browser window opens automatically. (I have already logged in to flickr earlier, so the browser takes me to my flick account.)

In the browser there is no instruction to follow: just open the main page of flickr.
At this point I answer with 'Y' at the command .uploadr.py and get:

```
Have you authenticated this application? (Y/N): È stata creata una nuova finestra nella sessione corrente del browser.
y
Error: {u'stat': u'fail', u'code': 108, u'message': u'Invalid frob'}
*****Removing empty Sets from DB*****
*****Completed removing empty Sets from DB*****
*****Adding Flickr Sets to DB*****
{'nojsoncallback': '1', 'format': 'json', 'perms': '', 'auth_token': 'None', 'api_sig': '******b3c9bdaa7d9877e7664185c960', 'api_key': '******19ae2e9c5add3e2fa4fb7bb6df', 'method': 'flickr.photosets.getList'}
Error: {u'stat': u'fail', u'code': 98, u'message': u'Invalid auth token'}
*****Completed adding Flickr Sets to DB*****
*****Uploading files*****
Found 1 files
Uploading /home/alfredo/Scaricati/flickr-uploader-master/foto/Selezione_001.png...
A problem occurred while attempting to upload the file: /home/alfredo/Scaricati/flickr-uploader-master/foto/Selezione_001.png
(<type 'exceptions.IOError'>, IOError('<?xml version="1.0" ?><rsp stat="fail">\n\t<err code="98" msg="Invalid auth token"/>\n</rsp>',), <traceback object at 0x7f5cbabc6c20>)
   1 files processed (uploaded, md5ed or timestamp checked)
*****Completed uploading files*****
*****Removing deleted files*****

```

The .flickr Token file contains a text: None
I have tried many times to resume the operation by deleting the files each time:
.flickrlock
.flickrToken
upload_flickr_db
But always with the same mistake.

I also have another old Flickr account on which in November 2016 (about) I successfully ran to uploadr.py

Last thing: I also tried to regenerate another apy-key, but I still have the same error.

Have they changed anything on flickr recently?

**Comment by alfem** [comment]
Same here. I think Flickr has changed some workflows

**Comment by ziegwald** [comment]
I tried it too, the URL just brought me to Flickr's home page :(

**Comment by volpik** [comment]
Same issue here, any solution?

**Comment by lodoss5** [comment]
I guess someone has to do some code changes to manage new way of gettting the token : https://www.flickr.com/services/api/auth.oauth.html

**Comment by jamesmstone** [comment]
I have done a similar thing [here](https://github.com/jamesmstone/flickr-uploadr/), also of relevance it's [#1](https://github.com/jamesmstone/flickr-uploadr/issues/1).

**Comment by alfem** [comment]
I have coded a quick script to to upload folders from my Synology to Flickr (using the new auth api). Tested in a DS212j. Perhaps it will be useful for any of you.

https://github.com/alfem/synology-flickr-folder-uploader

**Comment by ManuCart** [comment]
Same issue for me.
Is a solution exist?

**Comment by oPromessa** [comment]
Based on the suggestions above I'm converting this script to use flickrapi and the new Authentication (OAuth) method. Still work in progress but what I have so far can be found on [Overhaul flickr-uploader](https://github.com/oPromessa/flickr-uploader).

Again... still work in progress...

**Comment by oPromessa** [comment]
@ManuCart , @lodoss5, @ziegwald, @alfem , @cicciopizza 

## Solved!!! Version 2.4.1 is now out on https://github.com/oPromessa/flickr-uploader/

- New Authentication for flickr.
- Use of flickrapi.
- Multiprocessing to achieve speedy and parallel uploads. Marvellous for 59K photos Library like mine.
- A bunch of new options
- Setup instructions for Synology
- and much more

### **Do read the README.md**

Enjoy! And help out testing! I've applied to a secondary account on flickr.com for testing. 
You can do the same!

_Happy Halloween!_

**Comment by oPromessa** [comment]
I'm closing this topic due to lack of activity. Let us know should you have any further queries.
